packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -28,8 +52,11 @@ export default class Sayt extends LitElement {
   connectedCallback() {
     super.connectedCallback();
 
-    window.addEventListener(SAYT_EVENT.SAYT_SHOW, this.showSayt);
-    window.addEventListener(SAYT_EVENT.SAYT_HIDE, this.hideSayt);
+    window.addEventListener(SAYT_EVENT.SAYT_SHOW, this.showCorrectSayt);
+    window.addEventListener(AUTOCOMPLETE_RECEIVED_RESULTS_EVENT, this.showCorrectSayt);
+    window.addEventListener(SAYT_EVENT.SAYT_HIDE, this.hideCorrectSayt);
+    window.addEventListener('click', this.processClick);
+    window.addEventListener('keypress', this.processKeyPress);
```

Comment:
I think it's more usual to act on `keydown` for situations like these.

What's your opinion on having these listeners around only when sayt is visible?

Reply:
Two of them make SAYT visible, so those have to exist when SAYT is not visible. They technically don't need to exists when it is.

The others can presumably exist only in the opposite case. I'm unaware of what impact this would have on performance. I'm tending toward leaving this for now and optimizing later. Let's discuss this in person.

Reply:
Additionally, why use `keypress`? This at least guarantees that any `keydown` events will have fired. Is it a UX thing?

Reply:
`keypress` has been deprecated according to mdn: https://developer.mozilla.org/en-US/docs/Web/API/Document/keypress_event. 

They suggest using `keydown` or `beforeinput` instead.

Reply:
Thanks, @JoeyDeol, I didn't realize it was deprecated.